src/compiler/crystal/types.cr

```diff
@@ -312,6 +312,15 @@ module Crystal
       nil
     end
 
+    def superclasses
+      superclasses = [] of Type
+      klass = self
+      while klass = klass.superclass
+        superclasses << klass
+      end
+      superclasses
+    end
```

Comment:
BTW, I don't know if add `#superclasses` here is okay because this is useless in the whole compiler...
Or maybe this should just go into `src/compiler/crystal/macros/methods.cr` (or I can just drop this commit of `#superclasses`)